### PR TITLE
#1005 Do not remove page from delegate when adding page

### LINKED ISSUE #1005: SessionQuotaManagingDataStore should not use remove-then-add for delegate store
`SessionQuotaManagingDataStore` currently does two steps when adding a new page:

1. Remove existing page
2. Add new page

The first step is necessary for keeping the `SessionData` that tracks size/counts of pages per session up to date. The problem is that it also propagates this removal to the delegate store. In case of a remote Redis instance and a possibly asynchronous client, this can lead to all sorts of problems.

We fixed many rare bugs in our application when we switched to a custom version of `SessionQuotaManagingDataStore` that only propagates the removal when necessary.

### PR DESCRIPTION
**Pull request by theigl**
Resolves #1005.

### CODE REVIEW

File: datastores-parent/datastore-common/src/main/java/org/wicketstuff/datastores/common/SessionQuotaManagingDataStore.java [hunk 1]
```diff
@@ -153,10 +153,10 @@ synchronized void removeAllPages(IPageContext context, IPageStore delegate) {
 		/**
 		 * Removes a page by its identifier
 		 *
-		 * @param pageId The id of the page to remove
-		 * @return
```
Comment (solomax) [inline datastores-parent/datastore-common/src/main/java/org/wicketstuff/datastores/common/SessionQuotaManagingDataStore.java:157]: `@return` is necessary here to make javadoc happy :)

File: datastores-parent/datastore-tests/pom.xml [hunk 2]
```diff
@@ -76,5 +76,10 @@
 			<artifactId>websocket-jakarta-server</artifactId>
 			<scope>compile</scope>
 		</dependency>
+		<dependency>
+			<groupId>org.mockito</groupId>
+			<artifactId>mockito-core</artifactId>
+			<scope>compile</scope>
```
Comment (solomax) [inline datastores-parent/datastore-tests/pom.xml:82]: the scope shouldn't be `compile`
It should be removed (so `test` will be used)

File: datastores-parent/datastore-tests/src/main/java/org/wicketstuff/datastores/common/SessionQuotaManagingDataStoreTest.java [hunk 3]
```diff
@@ -1,6 +1,7 @@
 package org.wicketstuff.datastores.common;
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.*;
```
Comment (solomax) [inline datastores-parent/datastore-tests/src/main/java/org/wicketstuff/datastores/common/SessionQuotaManagingDataStoreTest.java:4]: Could you please `expand` this asterisk import? :))

**Review by solomax — CHANGES_REQUESTED** [review]

File: datastores-parent/datastore-tests/pom.xml (hunk 2 above)
Comment (theigl) [reply datastores-parent/datastore-tests/pom.xml:82]: `compile` scope is required at the moment. See the discussion in #1008

**Review by solomax — APPROVED** [review]

File: datastores-parent/datastore-common/src/main/java/org/wicketstuff/datastores/common/SessionQuotaManagingDataStore.java (hunk 1 above)
Comment (solomax) [reply datastores-parent/datastore-common/src/main/java/org/wicketstuff/datastores/common/SessionQuotaManagingDataStore.java:157]: @theigl could you please address this one? :)

Comment (theigl) [reply datastores-parent/datastore-common/src/main/java/org/wicketstuff/datastores/common/SessionQuotaManagingDataStore.java:157]: This is already addressed. For some reason this was not marked as outdated by GitHub.

**Comment by theigl** [comment]
@solomax: Please go ahead and merge this if it is OK for you. I don't seem to have commit rights to this repository (yet).

**Pull request merged**